packages/lib/src/interactions/utils.ts

```diff
@@ -1,7 +1,31 @@
-import type { ModifierKey } from './models';
+import type { SVGProps } from 'react';
+
+import type { ModifierKey, Rect } from './models';
 
 export function getModifierKeyArray(
   keys: ModifierKey | ModifierKey[] | undefined = []
 ): ModifierKey[] {
   return Array.isArray(keys) ? keys : [keys];
 }
+
+export function getSvgRectAttrs(rect: Rect): SVGProps<SVGRectElement> {
```

Comment:
I get why you used `SVGProps<SVGRectElement>` but it is a bit too broad to my taste.

I think dedicated types `{x: number, y: number, width: number, height: number}` would be much clearer for consumers for the function.

Also, you could rename the function to `getSvgRectCoords` or similar since it only returns coordinates.

Reply:
I went with the renaming but I chose to remove the return type altogether and let TS infer it. I had put a return type as a way to check that I wasn't returning invalid `rect` attributes from inside the function, but this is checked just as well when the function is used on a `rect` element. I could have used `Pick<SVGProps<...>, '...'>` but letting TypeScript do it's job seems much cleaner here.